src/business/dal/api/api.adapter.ts

```diff
@@ -0,0 +1,45 @@
+import IDataAdapter from '@business/dal/data-adapter.interface';
+import Api from './api.connection';
+import { Entity } from '../adapters/entities.list';
+import { apiEntitiesList } from './api-entities.list';
+
+export default class ApiAdapter implements IDataAdapter {
+  protected tableName: string;
+
+  constructor(entityKey: Entity) {
+    this.tableName = apiEntitiesList[entityKey];
```

Comment:
### 🤔 Question: What if apiEntitiesList doesn't have the `entityKey` key?

U need to do 2 things here:
1. Change `Entity` (u don't need to use it here any way) to some generic based on `apiEntitiesList`
2. Generate an exception if `apiEntitiesList[entityKey]` is undefined.

Reply:
Why shouldn't we use Entity here?
I think it's important to know what the key should be